### PR TITLE
Image Block: Update linkDestination when editing image link URL

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
## Description

There was a bug where once the media or attachment link destinations were set on an image you could not edit the image link URL without first having to click the clear button in the `ImageURLInputUI`. These changes fix that by updating the link destination as required when submitting the image link URL.

## How has this been tested?
Manually.

#### Testing Instructions

1. Add an image block to a post, select it and click the link icon in the block toolbar
2. Set the link to the "Media File" destination
3. Now click the edit button beside the link url input and attempt to edit the URL. It won't save.
4. Apply this PR branch
5. Reload the editor and repeat the process. It should now work as expected.

## Screenshots 

| Before | After |
|--------|------|
| EditingImageLink | EditingImageLinkFixed |

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by glendaviesnz — APPROVED** [review]
This tested well for me

File: packages/block-editor/src/components/url-popover/image-url-input-ui.js [hunk 1]
```diff
@@ -142,7 +142,15 @@ const ImageURLInputUI = ( {
 	const onSubmitLinkChange = useCallback( () => {
 		return ( event ) => {
 			if ( urlInput ) {
-				onChangeUrl( { href: urlInput } );
+				const selectedDestination =
+					getLinkDestinations().find(
+						( destination ) => destination.url === urlInput
+					)?.linkDestination || LINK_DESTINATION_CUSTOM;
```
Comment (glendaviesnz) [inline packages/block-editor/src/components/url-popover/image-url-input-ui.js:148]: This check is needed to account for instances where the url entered matches the media or attachment urls

Comment (aaronrobertshaw) [reply packages/block-editor/src/components/url-popover/image-url-input-ui.js:148]: That's correct 👍

**Comment by glendaviesnz** [comment]
@draganescu did you have any thoughts on this change?

**Review by apeatling — APPROVED** [review]
Worked well for me in testing.

**Pull request merged**